PhysicsTools/PatAlgos/plugins/PATTracksToPackedCandidates.cc

```diff
@@ -0,0 +1,203 @@
+
+  // ----------member data ---------------------------
+  const edm::EDGetTokenT<reco::TrackCollection> srcTracks_;
+  const edm::EDGetTokenT<reco::VertexCollection> srcPrimaryVertices_;
+  const double dzSigCut_;
+  const double dxySigCut_;
+  const double dzSigHP_;
+  const double dxySigHP_;
+  const double ptMax_;
+  const double ptMin_;
+  const bool resetHP_;
+  const int covarianceVersion_;
+  const int covarianceSchema_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+PATTracksToPackedCandidates::PATTracksToPackedCandidates(const edm::ParameterSet& iConfig)
+    : srcTracks_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("srcTracks"))),
+      srcPrimaryVertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("srcPrimaryVertices"))),
+      dzSigCut_(iConfig.getParameter<double>("dzSigCut")),
+      dxySigCut_(iConfig.getParameter<double>("dxySigCut")),
+      dzSigHP_(iConfig.getParameter<double>("dzSigHP")),
+      dxySigHP_(iConfig.getParameter<double>("dxySigHP")),
+      ptMax_(iConfig.getParameter<double>("ptMax")),
+      ptMin_(iConfig.getParameter<double>("ptMin")),
+      resetHP_(iConfig.getParameter<bool>("resetHP")),
+      covarianceVersion_(iConfig.getParameter<int>("covarianceVersion")),
+      covarianceSchema_(iConfig.getParameter<int>("covarianceSchema")) {
+  produces<std::vector<pat::PackedCandidate>>();
+  produces<edm::Association<pat::PackedCandidateCollection>>();
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void PATTracksToPackedCandidates::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+  using namespace std;
+
+  //track collection
+  auto tracks = iEvent.getHandle(srcTracks_);
+
+  auto outPtrTrksAsCands = std::make_unique<std::vector<pat::PackedCandidate>>();
+
+  //vtx collection
+  edm::Handle<reco::VertexCollection> pvs;
+  iEvent.getByToken(srcPrimaryVertices_, pvs);
+  reco::VertexRef pv(pvs.id());
+  reco::VertexRefProd pvRefProd(pvs);
+  if (!pvs->empty()) {
+    pv = reco::VertexRef(pvs, 0);
+  }
+
+  //best vertex
+  double bestvzError = -999.9, bestvxError = -999.9, bestvyError = -999.9;
+  const reco::Vertex& vtx = (*pvs)[0];
+  bestvzError = vtx.zError();
+  bestvxError = vtx.xError();
+  bestvyError = vtx.yError();
+  const math::XYZPoint& bestvtx(vtx.position());
+  math::Error<3>::type vtx_cov = vtx.covariance();
+
+  std::vector<int> mapping(tracks->size(), -1);
+  int savedCandIndx = 0;
+  int trkIndx = -1;
+  for (auto const& trk : *tracks) {
+    trkIndx++;
+    double dzvtx = std::abs(trk.dz(bestvtx));
+    double dxyvtx = std::abs(trk.dxy(bestvtx));
+    double dzerror = std::hypot(trk.dzError(), bestvzError);
+    double dxyerror = sqrt(trk.dxyError(bestvtx, vtx_cov) * trk.dxyError(bestvtx, vtx_cov) + bestvxError * bestvyError);
```

Comment:
```suggestion
    double dxyerror = trk.dxyError(bestvtx, vtx_cov);
```
IIUC, the vertex uncertainties are added to dxyError appropriately already.

Reply:
This comment is included in this commit:  https://github.com/cms-sw/cmssw/pull/30313/commits/7af9308247a1f43eab7529feab92d3720c1f3bbe